flipperdevices/flipperzero-firmware
# [FL-3332] Autolock fixes

# What's new
  - A desktop icon that shows up when a CLI session is active
  - Autolock inhibition when a CLI session is active (that includes RPC-over-USB sessions) - fixes FL-3332 and #3296
  - Always locks the device on boot if a PIN code is set irregardless of the RTC "Locked" flag - fixes #3900
  - Adds a crossed out lock icon to tell the user that autolocking is currently inhibited

# Verification 
  - Set up a PIN and 30s autolocking

Fix A:
  - While the device is unlocked, open a CLI session
  - While the device is still unlocked, press the back button to reset the inactivity timer, start a stopwatch at the same exact moment
  - Verify that the CLI session is still active after 30 seconds
  - Verify that there's a CLI icon on the desktop
  - Terminate the CLI session
  - Verify that the device locks itself after 30 seconds

Fix B:
  - Unlock the device, press the back button to reset the inactivity timer, start a stopwatch at the same exact moment
  - While the device is still unlocked, run `./fbt flash_usb_full` such that the process of uploading the files crosses the 30-second mark (the timing is tricky to get right, i admit)
  - Verify that the upload is not interrupted
  
Fix C:
  - Unlock the device, press the back button to reset the inactivity timer, start a stopwatch at the same exact moment
  - Connect to the device via qFlipper or Flipper Lab
  - While the device is still unlocked, update the firmware such that the process of uploading the files crosses the 30-second mark (again, kinda tricky)
  - Verify that the upload is not interrupted
  
Fix D:
  - Power-cycle the device using all available methods (soft reset, hard reset, reboot via settings, power off and on via settings)
  - Verify that the device locks itself upon startup no matter the state that it was in prior to it

# Checklist (For Reviewer)
